File: zero-rule-web/sub/domain/src/model/CLUser.ts
import { CLUserStateKey } from '../enum-types';

/**
 * @interface CLUser
 *
 * 사용자
 */
export interface CLUser {
  /**
   * 사용자 아이디
   */
  userId: number;

  /**
   * 로그인 아이디
   */
  lgonId: string;

  /**
   * 비밀번호
   */
  pswd: string;

  /**
   * 사용자 이름
   */
  userNm: string;

  /**
   * 생년월일
   */
  bryyMndy?: string;

  /**
   * 사용자핸드폰번호
   */
  userTno?: string;

  /**
   * 사용자내선번호
   */
  userExno?: string;

  /**
   * CTI사용자고유번호
   */
  ctiUserNtno?: string;

  /**
   * 프린트가능여부
   */
  prtPosbYn: 'Y' | 'N';

  /**
   * 다운로드가능여부
   */
  dwnlPosbYn: 'Y' | 'N';

  /**
   * 야간여부
   */
  atntYn: 'Y' | 'N';

  /**
   * 팀아이디
   */
  teamId?: number;

  /**
   * 사용자등급
   */
  userGd: string;

  /**
   * 사용자 상태
   */
  userState: CLUserStateKey;

  /**
   * 로그인 실패 횟수
   */
  lgonFailCnt: number;

  /**
   * 비밀번호 초기화여부
   */
  pswdInitYn: 'Y' | 'N';

  /**
   * 최종비밀번호변경일시
   */
  lastPswdChngDttm?: number;

  /**
   * 최종 로그인 일시
   */
  lastLgonDttm?: number;

  /**
   * 메모
   */
  memo?: string;

  /**
   * 생성일
   */
  regDttm: number;

  /**
   * 등록 사용자 ID
   */
  regrId?: string;

  /**
   * 수정일
   */
  chngDttm: number;

  /**
   * 변경 사용자 ID
   */
  chgrId?: string;

  /**
   * 팀 이름
   */
  teamNm?: string;
}

export interface CLUserCreateParam {
  lgonId: string;
  userName: string;
  userState: CLUserStateKey;
  regrId?: string;
  bryyMndy?: string;
  userTno?: string;
  userExno?: string;
  userGd?: string;
  ctiUserNtno?: string;
  prtPosbYn?: 'Y' | 'N';
  dwnlPosbYn?: 'Y' | 'N';
  atntYn?: 'Y' | 'N';
  memo?: string;
  teamId?: number;
}

File: zero-rule-server-main/domain/src/main/java/server/domain/entity/CLUser.java
package server.domain.entity;

import cl.cloverframework.ICLUser;
import cl.cloverframework.log.CLUserState;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.lang.NonNull;
import org.springframework.lang.Nullable;

import java.time.Instant;

/**
 * CL 사용자
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CLUser implements ICLUser {

	/**
	 * PK
	 * 사용자ID
	 */
    @NonNull
	private long userId;

	/**
	 * 로그인 ID
	 */
	@NonNull
	private String lgonId;

	/**
	 * 비밀번호
	 */
	@NonNull
	private String pswd;

	/**
	 * 사용자 이름
	 */
	@Nullable
	private String userNm;

    /**
     * 생년월일
     */
    @Nullable
    private String bryyMndy;

    /**
     * 사용자핸드폰번호
     */
    @Nullable
    private String userTno;

    /**
     * 사용자내선번호
     */
    @Nullable
    private String userExno;

    /**
     * CTI사용자고유번호
     */
    @Nullable
    private String ctiUserNtno;

    /**
     * 프린트가능여부
     */
    @Nullable
    private String prtPosbYn;

    /**
     * 다운로드가능여부
     */
    @Nullable
    private String dwnlPosbYn;

    /**
     * 야간여부
     */
    @Nullable
    private String atntYn;

    /**
     * 팀아이디
     */
    @Nullable
    private String teamId;

    /**
     * 사용자등급
     * ex) 0, 1, 2
     */
    @NonNull
    private String userGd;

	/**
	 * 사용자 상태
	 */
	@NonNull
	private CLUserState userState;

    /**
     * 로그인실패건수
     */
    @NonNull
    private long lgonFailCnt;

    /**
     * 비밀번호 초기화여부
     */
    @Nullable
    private String pswdInitYn;

    /**
     * 최종비밀번호변경일시
     */
    @Nullable
    private Instant lastPswdChngDttm;

	/**
	 * 최종 로그인일시
	 */
	@Nullable
	private Instant lastLgonDttm;

    /**
     * 메모
     */
    @Nullable
    private String memo;

    /**
     * 등록일시
     */
    @NonNull
    private Instant regDttm;

	/**
	 * 등록 사용자ID
	 */
	@Nullable
	private String regrId;

    /**
     * 변경일시
     */
    @NonNull
    private Instant chngDttm;

	/**
	 * 변경 사용자ID
	 */
	@Nullable
	private String chgrId;
}
